tasks/eva_3033/detect_project_status.py

```diff
@@ -0,0 +1,228 @@
+        yield current_analysis, current_assembly, current_tax_id, filenames
+
+    def get_taxonomy_for_project(self):
+        taxonomies = []
+        query = f"select distinct taxonomy_id from evapro.project_taxonomy where project_accession='{self.project}'"
+        for tax_id, in get_all_results_for_query(self.metadata_conn, query):
+            taxonomies.append(tax_id)
+        if len(taxonomies) == 1:
+            return taxonomies[0]
+        else:
+            logger.error(f'Cannot retrieve a single taxonomy for project {self.project}. Found {len(taxonomies)}.')
+
+    def check_accessioning_was_done(self, analysis, filenames):
+        """
+        Check that an accessioning file can be found in either noah or codon (assume access to both filesystem)
+        It parses and provide a 1000 submitted variant accessions from that project.
+        """
+        accessioning_reports = self.get_accession_reports_for_study()
+        accessioned_filenames = [self.get_accession_file(f) for f in filenames]
+        if not accessioning_reports:
+            return 0, []
+        if len(accessioning_reports) == 1:
+            accessioning_report = accessioning_reports[0]
+        elif len([r for r in accessioning_reports if r in accessioned_filenames]) == 1:
+            accessioning_report = [r for r in accessioning_reports if os.path.basename(r) in accessioned_filenames][0]
+        elif len([r for r in accessioning_reports if analysis in r]) == 1:
+            accessioning_report = [r for r in accessioning_reports if analysis in r][0]
+        elif accessioning_reports:
+            logger.warning(f'Assume all accessioning report are from project {self.project}:{analysis} and only use the first one.')
+            accessioning_report = accessioning_reports[0]
+        else:
+            logger.error(f'Cannot assign accessioning report to project {self.project} analysis {analysis} for files {accessioning_reports}')
+            accessioning_report = None
+        if not accessioning_report:
+            return 0, []
+        return self.get_accessioning_info_from_file(accessioning_report)
+
+    def get_accession_file(self, filename):
+        basefile = ''
+        if filename.endswith('.vcf.gz'):
+            basefile = filename[:-7]
+        elif filename.endswith('.vcf'):
+            basefile = filename[:-4]
+        return basefile + '.accessioned.vcf.gz '
+
+    def get_accessioning_info_from_file(self, path):
+        """
+        Read the accessioning report to retrieve the first 1000 Submitted variant accessions
+        """
+        no_of_ss_ids_in_file = 0
+        first_1000_ids = []
+        with gzip.open(path, 'rt') as f:
+            for line in f:
+                if line.startswith('#'):
+                    continue
+                elif line.split("\t")[2].startswith("ss"):
+                    no_of_ss_ids_in_file = no_of_ss_ids_in_file + 1
+                    if no_of_ss_ids_in_file <= 1000:
```

Comment:
I am wondering why retrieve 1000? is it to state with reasonable confidence that we have fully remapped everything? I am wondering if it would be better to look at sort by accession and look at first few and the last few IDs.

Reply:
This is a fair point, is there a chance that for example the file is sorted by contig and some contigs got remapped more successfully than others?

On the other hand I think this is intended more as a quick-and-dirty heuristic, so maybe we don't want to spend too much effort optimising our sampling method...